opal/corelib/file.rb

```diff
@@ -30,6 +30,12 @@ def basename(path)
       split(path)[-1]
     end
 
+    def extname(path)
+      return '' if path.nil_or_empty?
```

Comment:
Good catch, I though this was a standard Ruby method but in fact this is an Asciidoctor core extension [nil_or_empty.rb](https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/core_ext/object/nil_or_empty.rb).
Do you want me to add `nil_or_empty` extension or replace the line with a standard Ruby code ?